File: Practice Arrays/3Take discount or Not.py
t = int(input())

while t > 0:
    n, x, y = map(int, input().split())
    a = list(map(int, input().split()))
    t -= 1
    # Your code goes here
    total_without_discount=0
    total_with_discount=x
    for el in a:
        total_without_discount+=el
        total_with_discount+=(el-y) if (el-y>=0) else 0
    if(total_with_discount<total_without_discount):
        print("COUPON")
    else:
        print("NO COUPON")

File: Practice Arrays/codigo_cs/3Take discount or Not.cs
using System;

public class Test3
{
	public static void DiscountOrNot()
	{
		// your code goes here
        int t = int.Parse(Console.ReadLine());
        for (int _ = 0; _ < t; _++)
        {
            string[] inputs = Console.ReadLine().Split();
            int n = int.Parse(inputs[0]);
            int x = int.Parse(inputs[1]);
            int y = int.Parse(inputs[2]);

            int[] a = Array.ConvertAll(Console.ReadLine().Split(), int.Parse);

            int totalWithoutDiscount = 0;
            int totalWithDiscount = x;

            foreach (int el in a)
            {
                totalWithoutDiscount += el;
                totalWithDiscount += (el - y) >= 0 ? (el - y) : 0;
            }

            if (totalWithDiscount < totalWithoutDiscount)
            {
                Console.WriteLine("COUPON");
            }
            else
            {
                Console.WriteLine("NO COUPON");
            }

        }
	}
}
